src/floris/tools/floris_interface.py

```diff
@@ -71,30 +181,207 @@ def create_floris(self) -> None:
+            flow_field_dict["wind_shear"] = wind_shear
+        if wind_veer is not None:
+            flow_field_dict["wind_veer"] = wind_veer
+        if specified_wind_height is not None:
+            flow_field_dict["reference_wind_height"] = specified_wind_height
+        if air_density is not None:
+            flow_field_dict["air_density"] = air_density
+        if wake is not None:
+            self.floris.wake = wake
+        if wind_layout:
+            pass  # TODO: will need for heterogeneous flow
+        if turbulence_intensity is not None:
+            pass  # TODO: this should be in the code, but maybe got skipped?
+        if turbulence_kinetic_energy is not None:
+            pass  # TODO: not needed until GCH
+        if with_resolution is not None:
+            # TODO: Update  whereever this goes
+            # TODO: This is the grid_resolution, so the xxGrid will have to be updated, except that grid_resolution is is Vec3, or int
+            # NOTE: probably not needed because this was for v2 Curl functionality, and we're adding a Curl solver
+            pass
+
+        self.floris.flow_field = FlowField.from_dict(flow_field_dict)
+        if reinitialize_farm:
+            farm = farm = self.floris.farm._asdict()
+
+        x = self.floris.farm.layout_x
+        if layout_array is not None:
+            x, y = layout_array
+            # Recreating a Farm object to avoid errors if new and original layouts are different sizes
+            farm = self.floris.farm._asdict()
+            farm["layout_x"] = x
+            farm["layout_y"] = y
+            reinitialize_farm = True
+
+        turbine_id_check = len(x) != self.floris.farm.n_turbines
+        turbine_id_check &= len(self.floris.farm.turbine_map) > 1
+
+        # Check that turbine_id has bee provided if required
+        if turbine_id is None and turbine_id_check:
+            raise ValueError(
+                "`turbine_id` must be provided if the layout is changing and there are " "multiple turbine types!"
+            )
+
+        # If no value is assigned, then create turbine_id
+        if turbine_id is None and reinitialize_farm:
+            farm["turbine_id"] = [[*self.floris.farm.turbine_map][0]] * len(farm["layout_x"])
+            reinitialize_farm = True
+
+        # Assign the new turbine_id if one is input
+        if turbine_id is not None:
+            if farm == {}:
+                farm = self.floris.farm._asdict()
+            farm["turbine_id"] = turbine_id
+            reinitialize_farm = True
+
+        if wtg_id is None and reinitialize_farm:
+            if len(self.floris.farm.wtg_id) != len(farm["layout_x"]):
+                farm.pop("wtg_id")
+        if wtg_id is not None:
+            if farm == {}:
+                farm = self.floris.farm._asdict()
+            farm["wtg_id"] = wtg_id
+            reinitialize_farm = True
+
+        if reinitialize_farm:
+            farm["wind_directions"] = self.floris.flow_field.wind_directions
```

Comment:
So does farm expand the 1-D arrays of wind directions and wind speeds so that each wind direction has all the wind speeds?

Reply:
This is being used to update the wind_directions and wind_speeds in the farm if they were already updated in the flow_field, and if other variables were updated, and requiring the farm to be updated.